.github/workflows/upload_component.yml

```diff
@@ -0,0 +1,20 @@
+name: Push components to Espressif Component Service
+
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  upload_components:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+        with:
+          submodules: 'true'
```

Comment:
Don't we need `submodules: 'recursive'` here? 
I would suggest recursive here even if current components don't need it, because it can help avoiding Ooops in future

Reply:
fixed!